packages/core/types/index.js

```diff
@@ -1764,6 +1768,11 @@ export type BundlingProgressEvent = {|
   +phase: 'bundling',
 |};
 
+export type BundleGraphEvent = {|
+  +type: 'bundleGraph',
+  +bundleGraph: BundleGraph<NamedBundle>,
+|};
```

Comment:
What Devon meant in https://github.com/parcel-bundler/parcel/pull/8491#discussion_r1004751445 is that there could an internal event between PackagerRunner and ReporterRunner (which the reporter runner then just uses for setting the bundle graph but doesn't send it out to the reporter plugins).

But I think it's strange to have to split it up into two events in the first place. 
Why don't you use the same idea of passing either the bundlegraph ref or the bundlegraph object itself in the `packaging` event? The `PackagerRunnner` is already running on the main thread, so calling `report(something)` wouldn't perform any serialization anyway.